primitives/src/circuit/merkle_tree/mod.rs

```diff
@@ -75,10 +77,133 @@ where
 
     /// Enforce correct `merkle_proof` for the `elem` against
     /// `expected_merkle_root`.
-    fn enforce_merkle_proof(
+    fn enforce_membership_proof(
         &mut self,
         elem: Self::LeafVar,
         merkle_proof: Self::MerklePathVar,
         expected_merkle_root: Variable,
     ) -> Result<(), CircuitError>;
 }
+
+/// Gadget for the sparse merkle tree
+pub trait SparseMerkleTreeGadget<M>: MerkleTreeGadget<M>
+where
```

Comment:
please consider adding some example (testable) rustdoc, similar to doc for `MerkleTreeGadget`.

Reply:
Yes we should make sure to add doctests in our new PR. Done [here](https://github.com/EspressoSystems/jellyfish/pull/172/commits/1c64b57625d1db40ad59b2fe5d1d6e2554f8528b)